pkg/agent/multicast/multicast_route.go

```diff
@@ -0,0 +1,422 @@
+			outboundInterfaces[i] = c.multicastInterfaceConfigs[i].Name
+		}
+		err = c.addOutBoundMrouteEntry(msg.Src, msg.Dst, outboundInterfaces)
+		if err != nil {
+			klog.ErrorS(err, "Adding outbound mroute entry failed")
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *MRouteClient) runRouteClient(stopCh <-chan struct{}) {
+	klog.InfoS("Start running multicast routing daemon")
+	go func() {
+		for {
+			buf := make([]byte, 4096)
+			n, _ := syscall.Read(c.sockFD, buf)
+			if n > 0 {
+				c.queue.Add(string(buf[:n]))
+			}
+		}
+	}()
+
+	for i := 0; i < int(workerCount); i++ {
+		// Process multicast Group membership report or leave messages.
+		go wait.Until(c.worker, time.Second, stopCh)
+	}
+	<-stopCh
+	c.flushMRoute()
+}
+
+func (c *MRouteClient) worker() {
+	for c.processIGMPNocacheMsg() {
+	}
+}
+
+func (c *MRouteClient) addMrouteEntry(src net.IP, group net.IP, iif uint16, oifsVif []uint16) (err error) {
+	mc := &multicastsyscall.Mfcctl{}
+	origin := src.To4()
+	mc.Origin = [4]byte{origin[0], origin[1], origin[2], origin[3]}
+	g := group.To4()
+	mc.Mcastgrp = [4]byte{g[0], g[1], g[2], g[3]}
+	ttls := [32]byte{}
+	for _, v := range oifsVif {
+		ttls[v] = 1
+	}
+	mc.Ttls = ttls
+	mc.Parent = iif
+	return multicastsyscall.SetsockoptViMfcctl(c.sockFD, syscall.IPPROTO_IP, multicastsyscall.MRT_ADD_MFC, mc)
+}
+
+func (c *MRouteClient) delMrouteEntry(src net.IP, group net.IP, iif uint16) (err error) {
+	mc := &multicastsyscall.Mfcctl{}
+	origin := src.To4()
+	mc.Origin = [4]byte{origin[0], origin[1], origin[2], origin[3]}
+	g := group.To4()
+	mc.Mcastgrp = [4]byte{g[0], g[1], g[2], g[3]}
+	mc.Parent = iif
+	return multicastsyscall.SetsockoptViMfcctl(c.sockFD, syscall.IPPROTO_IP, multicastsyscall.MRT_DEL_MFC, mc)
+}
+
+func (c *MRouteClient) flushMRoute() {
+	klog.InfoS("Clearing multicast routing table entries")
+	multicastsyscall.SetsockoptVifctl(c.sockFD, syscall.IPPROTO_IP, multicastsyscall.MRT_FLUSH, &multicastsyscall.Vifctl{})
+}
+
+func (c *MRouteClient) deleteInboundMrouteEntryByGroup(group net.IP) (err error) {
+	klog.V(2).InfoS("Deleting multicast group", "group", group)
+	mEntries, _ := c.inboundRouteCache.ByIndex(GroupNameIndexName, group.String())
+	for _, route := range mEntries {
+		entry := route.(*inboundMulticastRouteEntry)
+		err := c.delMrouteEntry(net.ParseIP(entry.src), net.ParseIP(entry.group), entry.vif)
+		if err != nil {
+			return err
+		}
+		c.inboundRouteCache.Delete(route)
+	}
+	return nil
+}
+
+// addOutBoundMrouteEntry configures multicast route from Antrea gateway to oifs(outbound multicast interfaces)
+// to allow multicast senders in the pods send multicast traffic to external.
+func (c *MRouteClient) addOutBoundMrouteEntry(src net.IP, group net.IP, oifs []string) (err error) {
+	klog.InfoS("Adding outbound multicast route", "src", src, "group", group, "oifs", oifs)
+	vifs := make([]uint16, 0, len(oifs))
+	for _, oif := range oifs {
+		oifvif, err := c.vifAllocator.setVifIfNotExist(oif)
```

Comment:
oifVIF

Reply:
Updated and changed it to `oifVIF`. I thinks `vif` should be capitalized together because it abbreviates virtual network interface in kernel.   I have changed all the occurances to VIF except ones generated by cgo in the third_party directory